### DISCUSSION TITLE
How do I prevent reCAPTCHA from coming up?

### DISCUSSION THREAD

**Discussion by pantonov817** [ZFC-Digital/puppeteer-real-browser#187] [Q&A]
I want to prevent reCAPTCHA from coming up. 
I am rotating proxies and chrome profiles but captchas are still appearing.
The site is taxslayer.com
This is how i set up the browser.

```
var profile = getProfile('chrome', true);
const { connect } = require("puppeteer-real-browser");
var result = await connect({
	headless: false,
	args: [
		'--start-maximized',
		'--profile-directory=' + profile.name
	],
	customConfig: {
		userDataDir: profile.path
	},
	turnstile: true,
	connectOption: {
		defaultViewport: null					
	},
	disableXvfb: false,
	ignoreAllFlags: false,
	proxy:{host:'123.45.678.90',port:'1234'}
});
browser = result.browser;
```
Here is an example
![Screenshot 2024-10-01 125822](https://github.com/user-attachments/assets/b1ff504b-c26f-4318-a2fc-f5067040bd40)

**Comment by mdervisaygan** [comment]
https://github.com/zfcsoftware/puppeteer-real-browser?tab=readme-ov-file#httpsantcptcomscore_detector-recaptcha-v3-detects-me-what-should-i-do

You must use the ‘--user-data-dir=’ flag as specified in the documentation. Otherwise recaptcha v3 detects it. I will fix this problem with an update soon. 
I did not encounter captcha when I tested with this code. 

 ```
   const { browser, page } = await connect({
        headless: false,
        args: [
            '--start-maximized',
            `--user-data-dir=${__dirname}/zfc`
        ],
        customConfig: {},

        connectOption: {
            defaultViewport: null
        },
        disableXvfb: true,
    })

```
![image](https://github.com/user-attachments/assets/56b53afc-afc8-42cf-9471-a63f7bfce0e9)

**Reply by pantonov817** [reply]
It's not loading the profile this way. 
Which one is the correct one?
'--user-data-dir=C:\\Users\\panay\\AppData\\Local\\Google\\Chrome\\User Data\\Profile 1'
or
'--user-data-dir=C:\\Users\\panay\\AppData\\Local\\Google\\Chrome\\User Data'

**Reply by mdervisaygan** [reply]
> It's not loading the profile this way. Which one is the correct one? '--user-data-dir=C:\Users\panay\AppData\Local\Google\Chrome\User Data\Profile 1' or '--user-data-dir=C:\Users\panay\AppData\Local\Google\Chrome\User Data'

The first one is true. A different folder must be specified each time for profiling. Without adding this flag, recaptcha v3 detects it. I will make this update today or tomorrow.

**Reply by pantonov817** [reply]
Chrome is still not loading the profiles. Will look forward for the update. 
BR

**Comment by pantonov817** [comment]
@zfcsoftware Is there an update?

**Comment by RubberArchind** [comment]
I'm looking forward to this update too, I'm quite frustrated with Recaptcha V3

**Reply by codsec** [reply]
> I'm looking forward to this update too, I'm quite frustrated with Recaptcha V3

Did u find any solution for v3?

**Reply by RubberArchind** [reply]
I can get 0.7 score in antcpt.com when i add --user-data-dir arguments and set ignoreAllFlags to true
